sagemaker-python-sdk/tensorflow_script_mode_using_shell_commands/tensorflow_script_mode_using_shell_commands.ipynb

```diff
@@ -0,0 +1,303 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sagemaker\n",
+    "from sagemaker.tensorflow import TensorFlow\n",
+    "\n",
+    "estimator = TensorFlow(entry_point='launcher.sh',\n",
+    "                       dependencies=['deep_dream.py'],      # Including deep_dream.py as a dependency\n",
+    "                       train_instance_type='local',         # Run in local mode\n",
+    "                       train_instance_count=1,\n",
+    "                       role=sagemaker.get_execution_role(), # Passes to the container the AWS role that you are using on this notebook\n",
+    "                       framework_version='1.11.0',          # Uses TensorFlow 1.11\n",
+    "                       py_version='py3',\n",
+    "                       script_mode=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To start a training job, we call `estimator.fit(inputs)`, where inputs is a dictionary where the keys, named **channels**, have values pointing to the data location:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inputs = {'training': f'file://{data_dir}'}\n",
+    "\n",
+    "estimator.fit(inputs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`estimator.model_data` contains the S3 location where the contents of **/opt/ml/model**\n",
+    "were save as tar.gz file. Let's untar and download the model:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!aws s3 cp {estimator.model_data} model.tar.gz\n",
+    "!tar -xvzf model.tar.gz"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can see the resulting image now:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from IPython.display import Image\n",
+    "Image(filename='dream.png')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Training in SageMaker"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After you test the training job locally, upload the dataset to an S3 bucket so SageMaker can access the data during training:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sagemaker\n",
+    "\n",
+    "training_data = sagemaker.Session().upload_data(path='training', key_prefix='datasets/deep-dream')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The returned variable inputs above is a string with a s3 location which SageMaker Tranining has permissions\n",
+    "to read data from. **It has education purposes, requiring\n",
+    " more robust solutions for larger datasets:**"
```

Comment:
* capitalize "S3"
* I'd change the first sentence to something like "The `upload_data` call above returns an S3 location that can be used during the SageMaker Training Job"
* I don't know what you're trying to say with the bold sentence